### DISCUSSION TITLE
'Simple' smoothed camera rotation / zooming about a fixed object?

### DISCUSSION THREAD

**Discussion by owdoo** [ramokz/phantom-camera#238] [Question / Help Wanted]
I saw the post about RTS controls, and I have a similar requirement, though simpler. 

My game idea involves small 3D islands, and I'd like to be able to rotate the camera around them, and zoom in and out. The rotation is fixed, and there's no lateral movement of the camera at all. The rotation will be done by clicking and dragging the mouse left or right, and zooming by using the scroll wheel (with suitable alternatives for keyboard, gamepad and touch).

I'd like short smooth lerping at the start and end of rotation / zooming, like Phantom Camera provides.

Do you think it would be possible to do this? Would it be done by placing an invisible object on the island centre and rotating that, and setting the camera to follow?

(FYI: I'm a bit of a beginner to 3D, having only done 2D in Godot 3.x a few years ago. I've recently been learning Unity, and have actually built a rough prototype of my idea from scratch, but I'm not happy with Unity's project/game file sizes, build times, C#, and corporate shenanigans! I'm sure I could muddle through in Godot 4.x, but Phantom Camera looks so good I'd love to use it for this and future projects)

**Comment by ramokz** [comment]
You could use a `Third Person` `Follow Mode`, which effectively makes the camera orbit around its target and allows for “zooming” towards and away from the target.

Then it's mainly just a matter of applying interpolation, or animating, the values of the various properties.
In this case, you would need to adjust the `Spring Length`, to change the “zoom”, and the rotation of the `SpringArm3D` to change the orbit position around the target.

For the `Spring Length` property, you can adjust that using [the setters](https://phantom-camera.dev/follow-modes/third-person#spring-length) and for the orbiting you can use either of the [two helper methods](https://phantom-camera.dev/follow-modes/third-person#third-person-rotation).
Again, to make them smooth, you can, e.g., `lerp` the values instead of just incrementing or decrementing the values directly.
